aikau/src/test/resources/testApp/js/aikau/testing/mockservices/DownloadArchiveMockXhr.js

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright (C) 2005-2015 Alfresco Software Limited.
+ *
+ * This file is part of Alfresco
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ *
+ * @module aikauTesting/DownloadArchiveMockXhr
+ * @author Dave Draper
+ */
+define(["dojo/_base/declare",
+        "dojo/_base/lang",
+        "dojo/_base/array",
+        "dojo/io-query",
+        "aikauTesting/MockXhr"], 
+        function(declare, lang, array, ioQuery, MockXhr) {
+   
+   return declare([MockXhr], {
+
+      /**
+       * This sets up the fake server with all the responses it should provide.
+       *
+       * @instance
+       */
+      setupServer: function alfresco_testing_DownloadArchiveMockXhr__setupServer() {
+         try
+         {
+            this.server.respondWith("POST",
+                                    "/aikau/proxy/alfresco/api/internal/downloads",
+                                    lang.hitch(this, this.createArchive));
+            this.server.respondWith("GET",
+                                    /\/proxy\/alfresco\/api\/internal\/downloads\/workspace\/(.*)/,
+                                    lang.hitch(this, this.reportProgress));
+            this.server.respondWith("DELETE",
+                                    /\/proxy\/alfresco\/api\/internal\/downloads\/workspace\/(.*)/,
+                                    lang.hitch(this, this.deleteArchive));
+         }
+         catch(e)
+         {
+            this.alfLog("error", "The following error occurred setting up the mock server", e);
+         }
+         this.alfPublish("ALF_MOCK_XHR_SERVICE_READY", {});
+      },
+
+      /**
+       * Simulates the initialization of archive generation.
+       *
+       * @instance
+       */
+      createArchive: function alfresco_testing_DownloadArchiveMockXhr__createArchive(request) {
+         var body = JSON.parse(request.requestBody);
+
+         // This widget is only intended to service a single request at a time, so we're just going
+         // to store the number of nodes as an instance variable...
+         this.bodyCount = body.length;
+
+         this.startTime = new Date().getTime();
```

Comment:
Well, yes... but it's only test code and I went for least common denominator in case we ever need to manually test against IE8 (where Date.now isn't supported)